target/linux/ath79/dts/qca9533_plasmacloud_pa300.dtsi

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qca953x.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	chosen {
+		/delete-property/ bootargs;
+	};
+
+	aliases {
+		led-boot = &led_status_green;
+		led-failsafe = &led_status_green;
+		led-running = &led_status_green;
+		led-upgrade = &led_status_green;
+		label-mac-device = &eth0;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		pinctrl-names = "default";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		status_red {
+			label = "red:status";
+			gpios = <&gpio 0 GPIO_ACTIVE_HIGH>;
+		};
+
+		led_status_green: status_green {
+			label = "green:status";
+			gpios = <&gpio 2 GPIO_ACTIVE_HIGH>;
+		};
+
+		status_blue {
+			label = "blue:status";
+			gpios = <&gpio 3 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	watchdog {
+		compatible = "linux,wdt-gpio";
+		gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+		hw_algo = "toggle";
+		/* hw_margin_ms is actually 300s but driver limits it to 60s */
+		hw_margin_ms = <60000>;
+		always-running;
+	};
+};
+
+&uart {
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+	num-cs = <0>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		/* partitions are passed via bootloader */
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x040000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x040000 0x040000>;
+			};
+
+			partition@80000 {
+				label = "custom";
+				reg = <0x080000 0x140000>;
+				read-only;
+			};
+
+			partition@1c0000 {
+				label = "inactive";
+				reg = <0x1c0000 0x700000>;
+			};
+
+			partition@8c0000 {
+				label = "inactive2";
+				reg = <0x8c0000 0x700000>;
+			};
+
+			art: partition@fc0000 {
+				label = "ART";
+				reg = <0xfc0000 0x040000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&eth0 {
+	status = "okay";
+	phy-handle = <&swphy4>;
+	mtd-mac-address = <&art 0x0>;
+};
+
+&eth1 {
+	/* Workaround: keep the Ethernet interfaces order/mapping correct
+	 * (GMAC0 -> eth0, GMAC1 -> eth1, same as in old ar71xx target)
+	 */
```

Comment:
I'm personally not a fan of this hack, but I think it got already merged at least once?

Reply:
I've copied this workaround from

* qca9531_alfa-network_r36a.dts
* qca9531_wallys_dr531.dts
* qca9533_plasmacloud_pa300.dtsi

Reply:
Yes, I also didn't like it back then ;-)

Reply:
@adschm any better idea how to handle broken/wrong/weird* interfaces assignment at the moment (in future I suppose we would have support for aliases)?

* you name it

Reply:
I have learned to live with it. Since we currently have part of the devices "correct" and part of the devices "swapped", I don't really see it necessary to change that ratio of these sets from 65:35 to 70:30.

To me, the workaround just seems like an additional step that is not really necessary, as I've accepted that the ethX assignment is arbitrary (not random) eventually, and config works either way.

Reply:
@adschm 

> I have learned to live with it.

Many users would also need to learn that when the next release is out.
Lets just hope they won't learn it the hard way.

> Since we currently have part of the devices "correct" and part of the devices "swapped", I don't really see it necessary to change that ratio of these sets from 65:35 to 70:30.

Plus extra Wiki page for migration and people complaining about swapped ethX interfaces mapping. If I was just an external user, I would say "good job guys"! We should have made the ratio XXX:0.

I have a very strict opinion here. Either we say that `ath79` is a brand new target without a clean and safe way for migration from `ar71xx` and we don't provide it at all (like start it from scratch) or we say `ath79` is a direct successor of the ar71xx and users are free and safe to update to the new target whenever their device gets support there. What we did in practice so far is something in between, IMHO very confusing from end user and downstream projects perspectives.

> To me, the workaround just seems like an additional step that is not really necessary, as I've accepted that the ethX assignment is arbitrary (not random) eventually, and config works either way.

Single line in DTS (with a comment so it won't get lost when this is *fixed*) and no headache with migration/update. And in fact, the ethX assignment is not arbitrary and it never was, even on the ar71xx (it was up to mach-* file author which interface got registered first during init - most were using that to register physical LAN as the eth0, that was kind of general convention... also kind of forgotten or abandoned now).

> But I won't stop you from keeping it, of course.

I always try to keep our users happy more than devs about the hacks and workarounds in the code (please, don't take that personally) ;)

We didn't come to a conclusion on this workaround (I remember asking about it on the ml) and nobody proposed a convenient (from end user perspective) solution for the problem at that time so we ended up with... Wiki page and bunch of devices requiring clean updates between ar71xx and ath79 (and thus 19.07 -> whatever comes next). On the other side, we took care about such things like LEDs naming mappings but not essential wired network setup :man_facepalming:

Reply:
I will think about your points ...